### PR TITLE
Initial generator for Whisper translation

### PR DESCRIPTION
**Pull request by wtulloch**
- Updated generator/openai.py to include the means for handling calls to whisper translation
- Added basic unit tests for the generator
- updated record_replay/openapi.py to handle whisper translation response
- minor refactoring

### CODE REVIEW

File: src/aoai-api-simulator/src/aoai_api_simulator/app_builder.py [hunk 1]
```diff
@@ -164,7 +164,8 @@ async def catchall(request: Request):
                 return Response(status_code=500)
 
             # Apply limits here so that that they apply to record/replay as well as generate
-            if response.status_code < 300:
+            # Limits are not applied if the request uses form data (e.g. whisper translation)
+            if response.status_code < 300 and not context.is_form_data():
```
[the remark this answers is not in the document]
Comment (wtulloch) [reply src/aoai-api-simulator/src/aoai_api_simulator/app_builder.py:168]: Have done some work around this but probably need to discuss

File: src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py [hunk 2]
```diff
@@ -634,3 +634,97 @@ async def azure_openai_chat_completion(context: RequestContext) -> Response | No
...
+    content = text
+    completion_tokens = num_tokens_from_string(text, model_name)
+    if response_format == "json":
+        json_result = {"text": text}
+        content = json.dumps(json_result)
+
+    context.values[SIMULATOR_KEY_LIMITER] = "openai"
+    context.values[SIMULATOR_KEY_OPERATION_NAME] = "translation"
+    context.values[SIMULATOR_KEY_DEPLOYMENT_NAME] = deployment_name
+    context.values[SIMULATOR_KEY_OPENAI_PROMPT_TOKENS] = 0
```
[the remark this answers is not in the document]
Comment (wtulloch) [reply src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py]: Done (I Think)

File: src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py [hunk 3]
```diff
@@ -634,3 +634,97 @@ async def azure_openai_chat_completion(context: RequestContext) -> Response | No
...
+    is_match, path_params = context.is_route_match(
+        request=request, path="/openai/deployments/{deployment}/audio/translations", methods=["POST"]
+    )
+    if not is_match:
+        return None
+
+    _validate_api_key_header(context)
+
+    deployment_name = path_params["deployment"]
+    model_name = get_chat_model_from_deployment_name(context, deployment_name)
```
[the remark this answers is not in the document]
Comment (wtulloch) [reply src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py]: Done.

File: loadtest/loadtest_translations_1s_latency.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import logging
```
Comment (stuartleeks) [inline loadtest/loadtest_translations_1s_latency.py:1]: Leaving a comment as a reminder that I plan to do a couple of load test runs before merging

File: src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py [hunk 5]
```diff
@@ -125,32 +138,88 @@ def get_chat_model_from_deployment_name(context: RequestContext, deployment_name
...
+
     # Determine the target latency for the request
     completion_tokens = context.values.get(constants.SIMULATOR_KEY_OPENAI_COMPLETION_TOKENS)
-
     if completion_tokens and completion_tokens > 0:
-        config = context.config
-        operation_name = context.values.get(constants.SIMULATOR_KEY_OPERATION_NAME)
         target_duration_ms = None
-        if operation_name == "embeddings":
+        if operation_name == OPENAI_OPERATION_EMBEDDINGS:
```
Comment (lucashuet93) [inline src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py:187]: Constants are a nice add!

**Review by lucashuet93 — APPROVED** [review]
Looks good! Should test prior to merge.

**Review by stuartleeks — APPROVED** [review]

**Pull request merged**